src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -125,6 +127,8 @@ public ThreadPool(Settings settings) {
                 .put(Names.WARMER, settingsBuilder().put("type", "scaling").put("keep_alive", "5m").put("size", halfProcMaxAt5).build())
                 .put(Names.SNAPSHOT, settingsBuilder().put("type", "scaling").put("keep_alive", "5m").put("size", halfProcMaxAt5).build())
                 .put(Names.OPTIMIZE, settingsBuilder().put("type", "fixed").put("size", 1).build())
+                .put(Names.FETCH_SHARD_STARTED, settingsBuilder().put("type", "scaling").put("keep_alive", "5m").put("size", availableProcessors * 2).build())
+                .put(Names.FETCH_SHARD_STORE, settingsBuilder().put("type", "scaling").put("keep_alive", "5m").put("size", availableProcessors * 2).build())
```

Comment:
Ahh, I see the "unbounded queue, since we don't want to timeout, just let it execute based on how fast the local FS is)" now in the description, since you reschedule on rejection though, isn't that handled?